crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -477,22 +477,40 @@ pub struct QuiltStoreBlob<'a> {
 }
 
 impl<'a> QuiltStoreBlob<'a> {
+    /// Returns if an identifier is valid.
+    ///
+    /// Note an empty identifier is not valid.
+    pub fn is_valid_identifier(identifier: impl AsRef<str>) -> bool {
+        let id = identifier.as_ref();
+        id.starts_with(|c: char| c.is_alphanumeric()) && id == id.trim_end()
+    }
```

Comment:
Nit (very optional): I generally like the pattern of having a function like the following:
```rust
    pub fn check_identifier(identifier: impl AsRef<str>) -> Result<(), QuiltError> {
        let id = identifier.as_ref();
        if id.starts_with(|c: char| c.is_alphanumeric()) && id == id.trim_end() {
            Ok(())
        } else {
            Err(QuiltError::InvalidIdentifier(identifier))
        }
    }
```

This can then be used simply as 
```rust
        Self::check_identifier(&identifier)?;
```